applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -248,31 +374,74 @@ static void stream_configured_cb(struct bt_audio_stream *stream,
 		return;
 	}
 
+	ret = headset_pres_delay_find(CONFIG_BT_AUDIO_PRES_DLY_SRCH_MODE, channel_index,
+				      &new_pres_dly_us);
+	if (ret) {
+		LOG_ERR("Cannot get a valid presentation delay");
+		return;
+	}
+
 #if CONFIG_STREAM_BIDIRECTIONAL
 	if (ep_state_check(headsets[channel_index].sink_stream.ep,
 			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED) &&
 	    ep_state_check(headsets[channel_index].source_stream.ep,
 			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED))
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 	{
+		for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+			if (i != channel_index) {
+				if (headsets[i].headset_conn == NULL) {
+					continue;
+				}
+
+				ret = update_sink_stream_qos(&headsets[i], new_pres_dly_us);
+				if (ret) {
+					LOG_ERR("Presentation delay not set for %s headset: %d",
+						headsets[channel_index].ch_name, ret);
+				}
+			}
+		}
+
+		LOG_DBG("Set %s headset, connection %p, stream %p", headsets[channel_index].ch_name,
+			&headsets[channel_index].headset_conn,
+			&headsets[channel_index].sink_stream);
+
+		headsets[channel_index].sink_stream.qos->pd = new_pres_dly_us;
+
 		ret = bt_audio_stream_qos(headsets[channel_index].headset_conn, unicast_group);
 		if (ret) {
-			LOG_ERR("Unable to set up QoS for headset %d: %d", channel_index, ret);
```

Comment:
This is setting up the entire QoS for this headset, not just the presentation delay.